### DISCUSSION TITLE
How to install wheel from HuggingFace hub?

### DISCUSSION THREAD

**Discussion by gdahia** [python-poetry/poetry#8501] [Q&A]
I am trying to [add a HuggingFace `spacy` model as a dependency](https://huggingface.co/docs/hub/spacy#using-existing-models) in a `poetry` project. I tried the following
1. `poetry add https://huggingface.co/spacy/en_core_web_sm/resolve/main/en_core_web_sm-any-py3-none-any.whl`
2. Adding `en_core_web_sm = {url = "https://huggingface.co/spacy/en_core_web_sm/resolve/main/en_core_web_sm-any-py3-none-any.whl"}`
but none work.

However, running `poetry run python -m pip install https://huggingface.co/spacy/en_core_web_sm/resolve/main/en_core_web_sm-any-py3-none-any.whl` works perfectly.

Could you help? Thanks!

In case someone is curious why I want to install a `spacy` model from `huggingface`, the reason is that it is a model in a private registry, and `spacy` does not support those afaik.

**Comment by dimbleby** [comment]
ask hugging face to produce wheels that follow the standards I expect

https://packaging.python.org/en/latest/specifications/binary-distribution-format/?highlight=wheel%20filename#file-name-convention

**Reply by gdahia** [reply]
I am not sure I fully understand what you mean. Are you saying that if the whell followed the standard perfectly, then it would work as expected?

**Reply by dimbleby** [reply]
well anyway: if poetry failed to install a standards-compliant wheel then that would be a poetry bug.  But if hugging face produce broken wheels, then that is a hugging face bug.

**Reply by gdahia** [reply]
I agree that this is not a poetry bug, and sorry if my original message implied otherwise. I thought there might be some way to do this within poetry as `pip` installs it correctly.
